### ISSUE TITLE
Need a general a11y Dropdown component

### ISSUE THREAD

**Issue by eckdanny-osi** [ca-cwds/design-system#242]
The closest component we have in our assemblage of libs for _Dropdown_/_Menu_/_Flyout_/... is the `Dropdown` component from `reactstrap`. The sketch design artifacts feature the little _triangle_/_pip_ that is present in the `Popover` component, but the `Dropdown` is the more semantically appropriate choice and the behaviors/UX are better aligned with our use case. The design artifacts also (typically) use an icon as the semantic menu button. (e.g.; a vertical ellipsis that when activated shows a menu with menuitems)

There are some a11y concerns which need to be addressed (see [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-practices-1.1/#menu)) re: focus traps, keyboard navigation, etc. (the aria usage looks correct tho)

To sum up:

* Icon instead of std `Button` component supported
* Proper focus-trap and keyboard navigation
* Tweak styles (e.g.; pips, menuitem:focus, ...)

An issue has been opened on `reactstrap` re keyboard navigation, but we'll need to make a judgement call whether its worth it to implement our own version of this component or if we can get by with something less invasive.

**Comment by eckdanny-osi** [comment]
A [PR has been submitted](https://github.com/reactstrap/reactstrap/issues/1287) to `reactstrap` but it may not be merged. Why? The reactstrap project is a react-based abstraction over bootstrap4. Its aim is functional parity. But a few of the a11y concerns that were raised about our general Dropdown component needs are problematic in BS4, too. Our needs are not necessary aligned with the goals of reactstrap here. 🤔

Will chat with @dianaOSI to see if we can live with the current state. If not, we may have to support/maintain our own Dropdown.

**Comment by eckdanny-osi** [comment]
closed by #301
